README.md
# Many Speech Analyses

Welcome!

This project also lives on the [OSF](https://osf.io/3bmcp/).  
General project information can be found [here](https://many-speech-analyses.github.io).

## Brief summary

We recruited collaborators interested in forming data analysis teams to examine the same dataset.
The data analysis teams wrote a journal-ready methods and results section and then peer-reviewed each others analyses.
Next, we used meta-analytic techniques to explore the data analysis teams' findings. 

## Overview

Recent efforts to replicate published findings have uncovered surprisingly low success rates across disciplines. 
Moreover, several studies have highlighted the large degree of analytic flexibility in data analysis which can lead to substantially different conclusions based on the same data set. 
Thus, researchers have expressed their concerns that these researcher degrees of freedom might facilitate bias and can lead to claims that do not stand the test of time. 
Even greater flexibility is to be expected in fields in which the primary data lend themselves to a variety of possible operationalizations. 
The multidimensional, temporally extended nature of speech constitutes an ideal testing ground for assessing the variability in analytic approaches that derives not only from aspects of statistical modelling but also from decisions regarding the quantification of the measured behavior. 
In the present study, we gave the same speech production data set to 30 teams of researchers and asked them to answer the same research question. 
Using Bayesian meta-analytic tools, we have observed substantial variability between teams and submitted analyses, with analytic and researcher-related predictors having little or no effect on the reported effects.

## More...

The first MSA project paper is available as a pre-print on PsyArXiv. 
You can read it here: <https://psyarxiv.com/q8t2k/>. 

Want to play around with the data? 
We also have a shiny app available [here](https://jvcasillas.shinyapps.io/shiny_msa/). 


CHANGELOG.md
# Changelog

## 2022-08-17

- We have run a non-prereg meta-analytic model with `model_id` instead of `team` as a group-level effect, because many teams have submitted multiple analyses. This type of meta-model requires each individual estimate to be its own group-level effect. We also run a non-prereg model with `team:model_id` as a nested group-level effect.
- The team `nestor_idahoensis` specified Bayesian priors, but we were unsure which priors to use after scaling all the variables. So we opted for using the default priors of brms (flat priors for `b` and data-derived Student-*t* distribution for intercepts).
- We have excluded the number of post-hoc analyses (`n_phoc`) from the analytic and researcher-related model because the teams didn't report any as such.
- We have excluded the number of random effects (`n_rand`, i.e. conservativeness) from the analytic and researcher-related model because this variable is very much related to the estimate standard error (more random effects = higher standard error).
- Analytic and researcher-related predictors have been transformed to *z*-scores.

## 2022-06-16

- As mentioned below, We decided to keep to our original protocol in terms of refitting submitted models to get standardised effect sizes.
- However, we changed minor aspects of the protocol of the refitting procedure (see protocol at <https://many-speech-analyses.github.io/many_analyses/scripts/r/04_refit_workflow>), specifically file  and variable naming conventions and the use of treatment contrasts instead of sum coding.
- ~~AND, we will include two predictors in the meta-analytic model: typicality operationalisation (to account for categorical vs continuous) and outcome variable (to account for possible differences in sign of the effect depending on the outcome).~~ >> We decided not to include these, because these are included in the Analytic and researcher-related predictors model.

## 2022-05-27

- Sent email to AMPPS (AMPPS - Decision on Proposal ID AMPPS-21-0020.R1) asking to update the protocol.
- ~~David Sbarra says OK with option 3: We could redefine our effect size definition and use for example Westfall et al.'s proposal (2014) to define effect sizes in hierarchical models.~~ >> Afterwards, we decided not to redefine our effect size definition. We stick to the pre-registered protocol.

## 2022-04-29

- ~~Review round 1: some teams did not complete the review by the deadline. They will be able to do so later on.~~ >> We have skipped missing reviews since we were running late with the review process due to wrong calculation of time allowed for review (2 weeks per round instead of 1 week per round).
- We decided to drop last round of review with all analysis plus revision of review due to time.

## 2022-04-19

- Change to the peer-review questionnaire: now we ask for the teams' animal name instead of first and last name of reviewer.

## 2022-02-18

- New members are joining. Take note that they should do the intake form and their prior belief should be excluded from the analysis.
- Close teams by end of February (one month and half before analysis submission deadline).

## 2021-11-29

- Deleted question 3 from the analytic approach questionnaire.
- We merged the intake form, recruitment form, and analytical approach form into one as per <https://github.com/many-speech-analyses/many_analyses/issues/89>.

## 2021-10-19

- Changed wordings in <https://many-speech-analyses.github.io> after feedback from researchers.

## 2021-10-15

- Recruitment is done with Eventbrite, which makes emailing all participants easier.


scripts/r/00_sim_similarity.R
# betapart shenanigans --------------------------------------------------------


# Source libs and load simulated data -----------------------------------------

source(here::here("scripts", "r", "00_libs.R"))
sim_df <- read_csv(here("data", "sim", "simulated_df.csv"))
ma_sim_m5 <- readRDS(here("models", "sim", "ma_sim_m5.rds"))

# global color scheme / non-optimized
cat1_col = "#d95f02"
cat2_col = "#7570b3"
cat3_col = "#1b9e77"


# -----------------------------------------------------------------------------

# bring into betapart shape
fixed_df <- sim_df %>% 
  mutate(team_reviewer = paste0(team, "_", reviewer)) %>% 
  column_to_rownames(var = "team_reviewer") %>% 
  select(fixed_1,
         fixed_2,
         fixed_3,
         fixed_4,
         fixed_5)

random_df <- sim_df %>% 
  mutate(team_reviewer = paste0(team, "_", reviewer)) %>% 
  column_to_rownames(var = "team_reviewer") %>% 
  select(random_1,
         random_2,
         random_3,
         random_4,
         random_5)

# get betapart objects
df.core.fixed <- betapart.core(fixed_df)
df.core.random <- betapart.core(random_df)

# get pairwise distance
df.pair.fixed <- beta.pair(df.core.fixed, index.family = "sorensen")
df.pair.random <- beta.pair(df.core.random, index.family = "sorensen")

# which one is it? beta.sim / beta.sne / beta.sor?
# if beta.sor then

df.pair.fixed.sor <- broom::tidy(df.pair.fixed$beta.sor) %>% 
  rename(team = item1) %>% 
  group_by(team) %>% 
  summarise(distance_fixed_mean = mean(distance), .groups = "drop")

df.pair.random.sor <- broom::tidy(df.pair.random$beta.sor) %>% 
  rename(team = item1) %>% 
  group_by(team) %>% 
  summarise(distance_random_mean = mean(distance), .groups = "drop")

# this way we only get values for team1-15 ?
# this is how ecoRR does it

df.pair.fixed.sor <- as.matrix(beta.pair(df.core.fixed, index.family = "sorensen")$beta.sor)
fixed_similarity <- colMeans(df.pair.fixed.sor, na.rm = T) 

df.pair.random.sor <- as.matrix(beta.pair(df.core.random, index.family = "sorensen")$beta.sor)
random_similarity <- colMeans(df.pair.random.sor, na.rm = T)


# add to sim_df
sim_df_new <- sim_df %>% 
  mutate(fixed_similarity = fixed_similarity,
         random_similarity = random_similarity)

# store
write_csv(sim_df_new, here("data", "sim", "simulated_df_distances.csv"))

# plot -------------------------------------------------------------------------
  
# scale distances
sim_df_new <- sim_df_new %>% 
  mutate(fixed_similarity_s = scale(fixed_similarity),
         random_similarity_s = scale(random_similarity))

# plot 
fe_re <- ggplot(sim_df_new,
       aes(x = fixed_similarity_s,
           y = random_similarity_s)) +
  geom_vline(xintercept = 0, lty = "dashed") +
  geom_hline(yintercept = 0, lty = "dashed") +
  geom_point(alpha = 0.8, size = 3, pch = 21,
             fill = cat3_col) +
  geom_label_repel(aes(label = team),
                   box.padding   = 0.35, 
                   point.padding = 0.5,
                   segment.color = 'grey50') +
  labs(title = "Sørensen distance for fixed & random effect parameters",
       #subtitle = "",
       x = "\nDistance values for random effects",
       y = "Distance values for fixed effects\n") +
  theme_minimal()

es_fe <- left_join(sim_df_new, 
  spread_draws(ma_sim_m5, r_team[Team,], b_Intercept) %>% 
    mutate(team = Team, b_Intercept = r_team + b_Intercept) %>% 
    group_by(team) %>% 
    summarize(posterior_median = median(b_Intercept), .groups = "drop"), 
  by = "team") %>% 
  ggplot(., 
    aes(x = fixed_similarity_s, y = posterior_median, 
        fill = random_num, color = random_num)) + 
    geom_point(alpha = 0.8, size = 3, pch = 21) +
    geom_label_repel(aes(label = team),
                     box.padding   = 0.35, 
                     point.padding = 0.5,
                     fill = "white",
                     color = "black",
                     segment.color = 'grey50') + 
    scale_fill_viridis_c(option = "D", end = 1) +
    scale_color_viridis_c(option = "D", end = 1) +
    labs(title = "Effect size and Sørensen distance for fixed effects",
       #subtitle = "",
       y = "ES (posterior median)\n",
       x = "\n Distance values for fixed effects",
       fill = "# fixed effects",
       color = "# fixed effects") +
    theme_minimal()

es_re <- left_join(sim_df_new, 
  spread_draws(ma_sim_m5, r_team[Team,], b_Intercept) %>% 
    mutate(team = Team, b_Intercept = r_team + b_Intercept) %>% 
    group_by(team) %>% 
    summarize(posterior_median = median(b_Intercept), .groups = "drop"), 
  by = "team") %>% 
  ggplot(., 
    aes(x = random_similarity_s, y = posterior_median, 
        fill = fixed_num, color = fixed_num)) + 
    geom_point(alpha = 0.8, size = 3, pch = 21) +
    geom_label_repel(aes(label = team),
                   box.padding   = 0.35, 
                   point.padding = 0.5,
                   fill = "white",
                   color = "black",
                   segment.color = 'grey50') + 
    scale_fill_viridis_c(option = "D", end = 1) +
    scale_color_viridis_c(option = "D", end = 1) +
    labs(title = "Effect size and Sørensen distance for random effects",
       #subtitle = "",
       y = "ES (posterior median)\n",
       x = "\nDistance values for random effects",
       fill = "# fixed effects",
       color = "# fixed effects") +
    theme_minimal()

# TR: the patchwork of all three doesn't work for me, no idea why
fe_re + es_fe + es_re
